scripts/scr_add_man/scr_add_man.gml

```diff
@@ -100,131 +101,116 @@ function scr_add_man(argument0, argument1, argument2, argument3, argument4, argu
 	                if (loaded=0){
 	                    f=wop.escort_num[i];
 	                    if (obj_ini.ship_capacity[f]>obj_ini.ship_carrying[f]){
-	                        obj_ini.ship_carrying[f]+=1;obj_ini.lid[argument1,good]=f;loaded=1;
-	                        obj_ini.wid[argument1,good]=0;obj_ini.loc[argument1,good]=obj_ini.ship_location[f];
+	                        obj_ini.ship_carrying[f]+=1;obj_ini.lid[target_company,good]=f;loaded=1;
+	                        obj_ini.wid[target_company,good]=0;obj_ini.loc[target_company,good]=obj_ini.ship_location[f];
 	                    }
 	                }
 	            }
 	        }
         
         
         
-	        // wop=string_delete(argument9,0,4);
+	        // wop=string_delete(home_spot,0,4);
 	        // wop2=real(wop);
-	        /*obj_ini.lid[argument1,good]=wop2;
+	        /*obj_ini.lid[target_company,good]=wop2;
 	        obj_ini.loc=obj_ini.ship_location[wop2];
-	        obj_ini.wid[argument1,good]=0;*/
+	        obj_ini.wid[target_company,good]=0;*/
 	    }
-	    /*if (argument9="default") and (obj_ini.fleet_type!=1){// Need a more elaborate ship_carrying += here for the different types of units
-	        var first,backup;first=0;backup=0;i=0;
-	        repeat(30){i+=1;
-	            if (obj_ini.ship_class[i]="Battle Barge") and (first=0) and (obj_ini.ship_capacity[i]>obj_ini.ship_carrying[i]) then first=i;
-	            if (obj_ini.ship_class[i]="Strike Cruiser") and (backup=0) and (obj_ini.ship_capacity[i]>obj_ini.ship_carrying[i]) then backup=i;
-	        }
-	        if (first!=0){
-	            obj_ini.lid[argument1,good]=first;obj_ini.loc[argument1,good]=obj_ini.ship_location[first];
-	            obj_ini.wid[argument1,good]=0;obj_ini.ship_carrying[first]+=1;
-	        }
-	        if (first=0) and (backup!=0){
-	            obj_ini.lid[argument1,good]=backup;obj_ini.loc[argument1,good]=obj_ini.ship_location[backup];
-	            obj_ini.wid[argument1,good]=0;obj_ini.ship_carrying[backup]+=1;
-	        }
-	        if (first=0) and (backup=0){
-	            obj_ini.lid[argument1,good]=0;obj_ini.loc[argument1,good]="";obj_ini.wid[argument1,good]=0;exit;
-	        }
+	    obj_ini.role[target_company,good]=man_role;
+	    obj_ini.wep1[target_company,good]="";
+	    obj_ini.wep2[target_company,good]="";
+	    obj_ini.armor[target_company,good]="";
+	    obj_ini.chaos[target_company,good]=corruption;
+	    obj_ini.experience[target_company,good]=spawn_exp;
+	    obj_ini.spe[target_company,good]="";
+	    obj_ini.god[target_company,good]=0;
+		obj_ini.TTRPG[target_company, good] = {};
     
-	    }*/
-	    obj_ini.role[argument1,good]=argument0;
-	    obj_ini.wep1[argument1,good]="";
-	    obj_ini.wep2[argument1,good]="";
-	    obj_ini.armour[argument1,good]="";
-	    obj_ini.hp[argument1,good]=100;
```

Comment:
did you forget to copy this hp line, or is it handled in a different way now?

Reply:
its now handles within the struct